remme/tp/node_account.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 REMME
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import logging
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+from remme.protos.node_account_pb2 import (
+    NodeAccount,
+    NodeAccountMethod,
+    NodeAccountInternalTransferPayload,
+)
+
+from remme.shared.forms import NodeAccountInternalTransferPayloadForm
+
+from .basic import (
+    PB_CLASS,
+    PROCESSOR,
+    VALIDATOR,
+    BasicHandler,
+    get_data,
+)
+from remme.settings.helper import _get_setting_value
+
+
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'node_account'
+FAMILY_VERSIONS = ['0.1']
+
+
+class NodeAccountHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            NodeAccountMethod.TRANSFER_FROM_FROZEN_TO_UNFROZEN: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._transfer_from_frozen_to_unfrozen,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            }
+        }
+
+    def _transfer_from_frozen_to_unfrozen(self, context, node_account_public_key, internal_transfer_payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+        node_account = get_data(context, NodeAccount, node_account_address)
+
+        if node_account is None:
+            node_account = NodeAccount()
+
+        minimum_stake = int(_get_setting_value(context, 'remme.settings.minimum_stake'))
```

Comment:
If the value of `remme.settings.minimum_stake` is malformed, then this will result into a crash.

Reply:
@eugene-babichenko, done,  add `        
if minimum_stake < 0:
      raise InvalidTransaction('Wrong minimum stake address.')`, and add a unit test for it.

Reply:
Why here [atomic swap commision](https://github.com/Remmeauth/remme-core/blob/dev/remme/tp/atomic_swap.py#L167) just check for a 
non-negativity number? If that's enough, then I can too just check on non-negativity?

Reply:
See 
https://github.com/Remmeauth/remme-core/pull/233#discussion_r261522347

AS comission is to be fixed